lib/granite/performer_proxy/proxy.rb

```diff
@@ -13,7 +15,7 @@ def inspect
         "<#{@klass}PerformerProxy #{@performer}>"
       end
 
-      def method_missing(method, *args, &block)
+      ruby2_keywords def method_missing(method, *args, &block)
         if @klass.respond_to?(method)
           @klass.with_proxy_performer(@performer) do
             @klass.public_send(method, *args, &block)
```

Comment:
Can't we just replace this by `(method, *args, **options, &block)`, to be compatible with ruby 2.6 instead of using the `ruby2_keywords`?
```suggestion
      ruby2_keywords def method_missing(method, *args, **options, &block)
        if @klass.respond_to?(method)
          @klass.with_proxy_performer(@performer) do
            @klass.public_send(method, *args, **options, &block)
```

Reply:
@catks I tried your suggestion before coming up with this one. CI checks for ruby 2.6 were failing. That's why I decided to go with the current one.

Reply:
Do you remember what the issues were? Perhaps there was something wrong with the test? Because I'm pretty sure 2.6 should support passing `**options` right? And we really only use this proxy for `as(performer)`, so keyword arguments shouldn't even matter.

Reply:
Ok, so this fixes the issue for ruby `2.7` but it still breaks ruby 2.6 since keywords work differently when using `send`, eg:

```ruby
# WIth ruby 2.6
def test(value) 
end

options = {}
test(42, **options) # works
send(:test, 42, **options) # raises ArgumentError (wrong number of arguments (given 2, expected 1))
```

It is possible to 'fix' some of the granite specs by changing the method signature (eg: https://github.com/toptal/granite/pull/73/commits/8eefbe74973c233dcb19ab5a33bd2eab4900f305#diff-1b00a5d80953e2e89aeaab3cfe7a7f7eafaf0d2a006305be4064284b7b28a8f0R10) but personally, I don't think it's a good approach as we will force ruby 2.6 users to also update their methods when upgrading.

Also [one of the specs](https://github.com/toptal/granite/blob/9f90ea83bf3d614e6ab17e6176d872507e77b730/spec/lib/granite/projector_spec.rb#L46) would need a different implementation for ruby 2.6 since an `Action` also receives `#new` with `[*params, {}]` instead of [*params].

Maybe we could remove the **ruby2_keywords** approach after dropping ruby 2.6 support or as another option we could  do something like this:

```ruby
   if RUBY_ENGINE && Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7")
        def method_missing(method, *args, **options, &block)
          if @klass.respond_to?(method)
            @klass.with_proxy_performer(@performer) do
              @klass.public_send(method, *args, **options, &block)
            end
          else
            super
          end
        end
      else
        def method_missing(method, *args, &block)
          if @klass.respond_to?(method)
            @klass.with_proxy_performer(@performer) do
              @klass.public_send(method, *args, &block)
            end
          else
            super
          end
        end
      end
```

But I guess I would keep the ruby2_keywords since it's simpler, WDYT? @andriusch

Reply:
OK let's keep `ruby2_keywords` and drop support for ruby 2.6 once `ruby2_keywords` is removed.